### PR TITLE
[android] Properly show max editable building levels

### LINKED ISSUE #4309: Update strings about max level in editor
Recently we have increase max level about building https://github.com/organicmaps/organicmaps/pull/4215 but we have forgotten to update strings in editor

Update here for android https://github.com/organicmaps/organicmaps/blob/81c1d08024b950220c718ffab639251ea6dda00d/android/src/app/organicmaps/editor/EditorFragment.java#L425
Check on IOS

CC @biodranik

### PR DESCRIPTION
**Pull request by biodranik**
Fixes #4309

### CODE REVIEW

**Comment by Jean-BaptisteC** [comment]
Did you have check on IOS ?
Please update `[error_enter_correct_storey_number]` strings in https://github.com/organicmaps/organicmaps/blob/master/data/strings/strings.txt

**Comment by biodranik** [comment]
@Jean-BaptisteC thanks, great catch! I've fixed it. iOS is not affected, the code there is written properly, without hard-coding :)

Can you please recheck if the fix works? Do you also have an iOS device?

**Review by j13m126 — APPROVED** [review]

**Review by LaoshuBaby — APPROVED** [review]

File: data/strings/strings.txt [hunk 1]
```diff
@@ -13383,44 +13383,44 @@
...
+    id = Edit bangunan dengan maksimum %d lantai
+    it = Il numero di piani non deve superare %d
+    ja = 最高%d階までのビルを編集
+    ko = 최대 %d층까지 입력하세요
+    mr = मजल्यांची संख्या %d पेक्षा जास्त नसावी
+    nb = Rediger bygningen med maks. %d etasjer
+    nl = Bewerk het gebouw met maximaal %d verdiepingen
+    pl = Edytuj budynek z maksymalną liczbą %d pięter
+    pt = Editar o edifício com um máximo de %d pisos
+    pt-BR = O número de andares não pode ultrapassar %d
```
Comment (matheusgomesms) [inline data/strings/strings.txt:13413]: pt = O número de pisos não pode ser maior que %d
 pt-BR = O número de andares não pode ser maior que %d

**Review by matheusgomesms — CHANGES_REQUESTED** [review]

File: android/src/app/organicmaps/editor/EditorFragment.java [hunk 2]
```diff
@@ -177,7 +177,13 @@ public void onTextChanged(CharSequence s, int start, int before, int count)
...
       public void onTextChanged(CharSequence s, int start, int before, int count)
       {
-        UiUtils.setInputError(mInputBuildingLevels, Editor.nativeIsLevelValid(s.toString()) ? 0 : R.string.error_enter_correct_storey_number);
+        final Context context = mInputBuildingLevels.getContext();
+        final EditText editText = mInputBuildingLevels.getEditText();
+        final boolean isValid = Editor.nativeIsLevelValid(s.toString());
+        mInputBuildingLevels.setError(isValid == 0 ? null
+            : context.getString(R.string.error_enter_correct_storey_number, Editor.nativeGetMaxEditableBuildingLevels()));
+        editText.setTextColor(isValid == 0 ? ThemeUtils.getColor(context, android.R.attr.textColorPrimary)
+                                           : context.getResources().getColor(R.color.base_red));
```
Comment (vng) [inline android/src/app/organicmaps/editor/EditorFragment.java:186]: Better !isValid

File: android/src/app/organicmaps/editor/EditorFragment.java [hunk 3]
```diff
@@ -422,7 +428,7 @@ private void initViews(View view)
 
     // Details
     mBlockLevels = view.findViewById(R.id.block_levels);
-    mBuildingLevels = findInputAndInitBlock(mBlockLevels, 0, getString(R.string.editor_storey_number, 25));
+    mBuildingLevels = findInputAndInitBlock(mBlockLevels, 0, getString(R.string.editor_storey_number, Editor.nativeGetMaxEditableBuildingLevels()));
```
Comment (vng) [inline android/src/app/organicmaps/editor/EditorFragment.java:431]: nit: Split on 2 lines?

File: data/strings/strings.txt [hunk 4]
```diff
@@ -13383,44 +13383,44 @@
...
-    sv = Redigera byggnaden med max 25 våningar
-    th = แก้ไขอาคารที่มีจำนวนชั้นมากกว่า 25
-    tr = Kat sayısı 25'i geçmemelidir
-    uk = Редагуйте будівлі висотою максимум 25 поверхів
-    vi = Sửa tòa nhà có tối đa 25 tầng
-    zh-Hans = 编辑最多 25 层的建筑
-    zh-Hant = 編輯最多 25 層的建築
+    comment = Error message in Editor when a user tries to set the number of floors for a building higher than %d floors
+    tags = android,ios
+    en = The number of floors must non exceed %d
```
Comment (vng) [inline data/strings/strings.txt:13388]: If we added int param to this string, we should update iOS code?

**Review by MetehanOzyurek — APPROVED** [review]

File: data/strings/strings.txt (hunk 4 above)
Comment (biodranik) [reply data/strings/strings.txt:13388]: Int param was already used there :)

**Comment by biodranik** [comment]
@vng PTAL

File: data/strings/strings.txt (hunk 4 above)
Comment (vng) [reply data/strings/strings.txt:13388]: How it was already used if pervious string was without Int?

**Review by vng — APPROVED** [review]

**Comment by Jean-BaptisteC** [comment]
> @Jean-BaptisteC thanks, great catch! I've fixed it. iOS is not affected, the code there is written properly, without hard-coding :)
> 
> Can you please recheck if the fix works? Do you also have an iOS device?

Yes, i can test, but workflow failed :(

I don't have IOS device to test.

**Review by matheusgomesms — APPROVED** [review]

File: data/strings/strings.txt [hunk 5]
```diff
@@ -13383,44 +13383,44 @@
...
+    fi = Muokkaa kerroksia, enintään %d kerrosta
+    fr = Le nombre d'étages ne doit pas dépasser %d
+    hu = Legfeljebb %d emeletes épület szerkesztése
+    id = Edit bangunan dengan maksimum %d lantai
+    it = Il numero di piani non deve superare %d
+    ja = 最高%d階までのビルを編集
+    ko = 최대 %d층까지 입력하세요
+    mr = मजल्यांची संख्या %d पेक्षा जास्त नसावी
+    nb = Rediger bygningen med maks. %d etasjer
+    nl = Bewerk het gebouw met maximaal %d verdiepingen
```
Comment (jimcarst) [inline data/strings/strings.txt:13410]:
```suggestion
    nl = Het gebouw mag niet meer dan %d verdiepingen hebben
```

**Comment by Jean-BaptisteC** [comment]
OK for translations, but correction doesn't work OM show error_enter_correct_storey_number string always when value is between 1 and 50 and nothing when value is superior to 51.

**Comment by d4f5409d** [comment]
Please fix this. https://github.com/organicmaps/organicmaps/pull/4315#discussion_r1085559188

File: android/res/values-hu/strings.xml [hunk 6]
```diff
@@ -429,8 +429,8 @@
 	<string name="download_over_mobile_message">Ez jelentősen drága lehet némely előfizetés vagy roaming keretein belül.</string>
 	<string name="error_enter_correct_house_number">Helyes házszámot adjon meg</string>
 	<string name="editor_storey_number">Az emeletek száma (legfeljebb %d)</string>
-	<!-- Error message in Editor when a user tries to set the number of floors for a building higher than 25 -->
-	<string name="error_enter_correct_storey_number">Legfeljebb 25 emeletes épület szerkesztése</string>
+	<!-- Error message in Editor when a user tries to set the number of floors for a building higher than %d floors -->
+	<string name="error_enter_correct_storey_number">Legfeljebb %d emeletes épület szerkesztése</string>
```
Comment (d4f5409d) [inline android/res/values-hu/strings.xml:433]:
```suggestion
	<string name="error_enter_correct_storey_number">Ez az épület legfeljebb %d emeletes lehet</string>
```

**Review by d4f5409d — CHANGES_REQUESTED** [review]
Please consider changing the sentence I fixed.

**Review by Jean-BaptisteC — APPROVED** [review]
✅Pixel 6 -Android 13
Error message is show if value is equal to 0 or superior to 50

**Pull request merged**